src/validation.cpp

```diff
@@ -3426,6 +3426,11 @@ static FlatFilePos SaveBlockToDisk(const CBlock& block, int nHeight, const CChai
 /** Store block on disk. If dbp is non-nullptr, the file is known to already reside on disk */
 bool CChainState::AcceptBlock(const std::shared_ptr<const CBlock>& pblock, CValidationState& state, const CChainParams& chainparams, CBlockIndex** ppindex, bool fRequested, const FlatFilePos* dbp, bool* fNewBlock)
 {
+    // Ignore blockpause in regtest
```

Comment:
Why?

Reply:
Say you generate 100 blocks with 1000 blockpause. This would take 100 seconds and the program would hang up.

Reply:
That is not a valid concern imo, especially with the default value being 0. Regtest exists to test features on mainnet, and this would prevent testing of blockpause in functional tests for no pressing reason.

Reply:
You're right but this feature was meant for the block download, not creation